warn/warn_test.go

```diff
@@ -510,6 +510,118 @@ load(":b.bzl", "b")
 		}, scopeBuild|scopeBzl)
 }
 
+func TestUnsortedDictItems(t *testing.T) {
+	checkFindingsAndFix(t, "unsorted-dict-items", `
+d = {
+	"b": "b value",
+	"a": "a value",
+}`, `
+d = {
+	"a": "a value",
+	"b": "b value",
+}`,
+		[]string{":3: Dictionary items are out of their lexicographical order."},
+		scopeEverywhere)
+
+	checkFindingsAndFix(t, "unsorted-dict-items", `
+d = {
+	"a": "a value",
+	"a": "a value",
+}`, `
+d = {
+	"a": "a value",
+	"a": "a value",
+}`,
+		[]string{},
+		scopeEverywhere)
+
+	checkFindingsAndFix(t, "unsorted-dict-items", `
+d = {
+	2: "two",
+	"b": "b value",
+	1: "one",
+	"a": "a value",
+	3: "three",
+}`, `
+d = {
+	2: "two",
+	"a": "a value",
+	1: "one",
+	"b": "b value",
+	3: "three",
+}`,
+		[]string{":5: Dictionary items are out of their lexicographical order."},
+		scopeEverywhere)
+
+	checkFindingsAndFix(t, "unsorted-dict-items", `
+d = {}`, `
+d = {}`,
+		[]string{},
+		scopeEverywhere)
+
+	checkFindingsAndFix(t, "unsorted-dict-items", `
+d = {
+	# b comment
+	"b": "b value",
+	"a": "a value",
+}`, `
+d = {
+	"a": "a value",
+	# b comment
+	"b": "b value",
+}`,
+		[]string{":4: Dictionary items are out of their lexicographical order."},
+		scopeEverywhere)
+
+	checkFindingsAndFix(t, "unsorted-dict-items", `
+# @unsorted-dict-items
+d = {
+	"b": "b value",
+	"a": "a value",
+}`, `
+# @unsorted-dict-items
+d = {
+	"b": "b value",
+	"a": "a value",
+}`,
+		[]string{},
+		scopeEverywhere)
+
+	checkFindingsAndFix(t, "unsorted-dict-items", `
+d = {
+	"key" : {
+		"b": "b value",
+		"a": "a value",
+	}
+}`, `
+d = {
+	"key" : {
+		"a": "a value",
+		"b": "b value",
+	}
+}`,
+		[]string{"4: Dictionary items are out of their lexicographical order."},
+		scopeEverywhere)
+
+	checkFindingsAndFix(t, "unsorted-dict-items", `
+# @unsorted-dict-items
+d = {
+	"key" : {
+		"b": "b value",
+		"a": "a value",
+	}
+}`, `
+# @unsorted-dict-items
```

Comment:
Tip: you can just use `checkFindings` if the input and the output are the same. It also ensures that the code is not modified because fixes are not expected.